### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by kyriakosoikonomakos**
- Filter instances per stack when looking for master
- env.passwords is used internally in fabric.
- get_new_stack_name renamed to get_stack_name
- code requires SSL cert details even when SSL isn't used.

### CODE REVIEW

File: fabfile.py [hunk 1]
```diff
@@ -86,13 +83,15 @@ def get_config():
...
+        env.stack_passwords={}
     aws_config = AWSConfig(env.aws)
     project_config = ProjectConfig(
         env.config,
         env.environment,
-        passwords=env.passwords)
+        passwords=env.stack_passwords)
 
-    cfn_config = ConfigParser(project_config.config, env.stack_name)
+    cfn_config = ConfigParser(project_config.config, get_stack_name)
```
Comment (mattmb) [inline fabfile.py]: shouldn't that be get_stack_name()

File: fabfile.py [hunk 2]
```diff
@@ -270,7 +275,7 @@ def install_master():
     stack_ips.remove(master_prv_ip)
     stack_public_ips = ec2.get_instance_public_ips(instance_ids)
     stack_public_ips.remove(master_public_ip)
-
+    print master_public_ip
```
Comment (ashb) [inline fabfile.py]: Looks like a debug print left in here

File: bootstrap_cfn/ec2.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import sys
 import boto.ec2
-
+from awsutils import cloudformation
```
Comment (ashb) [inline bootstrap_cfn/ec2.py]: This should be `from bootstrap_cfn import cloudformation`

**Comment by kyriakosoikonomakos** [comment]
The test that failed needs to be fixed... the failure is on whitespace.

**Pull request merged**